### PR TITLE
Ping pong

### PR DESCRIPTION
**Pull request by mirpo**
- added `ping` / `pong` functions in broker
- fixed crashing `mqtt server` (whole broker), when we were trying to unsubscribe already disconnected `mqtt client`

### CODE REVIEW

File: test/integration/SubscriptionManager.test.js [hunk 1]
```diff
@@ -102,20 +98,24 @@ describe('SubscriptionManager', () => {
...
-
         expect(broker1.getStreams()).toEqual([freshStreamId1 + '::0', freshStreamId2 + '::0'].sort())
         expect(broker2.getStreams()).toEqual([freshStreamId1 + '::0', freshStreamId2 + '::0'].sort())
 
         await client1.subscribe({
             stream: freshStreamId1
         }, () => {})
 
-        await wait(2000)
+        await waitForCondition(() => broker1.getStreams().length === 2)
```
Comment (harbu) [inline test/integration/SubscriptionManager.test.js:108]: Hmm.. before it is already checked that
```js
expect(broker1.getStreams()).toEqual([freshStreamId1 + '::0', freshStreamId2 + '::0'].sort())
expect(broker2.getStreams()).toEqual([freshStreamId1 + '::0', freshStreamId2 + '::0'].sort())
```
which implies length == 2 for both broker1 and broker2.

File: src/websocket/WebsocketServer.js [hunk 2]
```diff
@@ -409,6 +426,26 @@ module.exports = class WebsocketServer extends EventEmitter {
...
+        const connections = [...this.connections.values()]
+        connections.forEach((connection) => {
+            try {
+                // didn't get "pong" in pingInterval
+                if (connection.isAlive !== undefined && !connection.isAlive) {
+                    throw Error('Connection is not active')
+                }
+
+                // eslint-disable-next-line no-param-reassign
+                connection.isAlive = false
```
Comment (harbu) [inline src/websocket/WebsocketServer.js:439]: I'd consider renaming `isAlive` to something like `respondedPong` or `pongResponded`. It's quite easy to confuse `isAlive = false` to mean that the connection is actually dead right now.

File: test/integration/ping-pong.test.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+const Publisher = require('../../src/Publisher')
+const VolumeLogger = require('../../src/VolumeLogger')
+const SubscriptionManager = require('../../src/SubscriptionManager')
+
+const trackerPort = 17370
+const httpPort = 17341
+const wsPort = 17351
+const networkNodePort = 17361
+
+describe('ping-ping test between broker and clients', () => {
```
Comment (harbu) [inline test/integration/ping-pong.test.js:17]: you mean ping-pong?

File: test/integration/ping-pong.test.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            pings += 1
+        })
+
+        // eslint-disable-next-line no-underscore-dangle
+        websocketServer._pingConnections()
+        await waitForCondition(() => pings === 3)
+
+        expect(pings).toEqual(3)
+
+        connections.forEach((connection) => {
```
Comment (harbu) [inline test/integration/ping-pong.test.js:92]: Might be worth checking `expect(connections.length).toEqual(3)` just to be on the safe side.

File: test/integration/ping-pong.test.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        client3.connection.socket.on('ping', () => {
+            pings += 1
+        })
+
+        // eslint-disable-next-line no-underscore-dangle
+        websocketServer._pingConnections()
+        await waitForCondition(() => pings === 2)
+
+        const connections = [...websocketServer.connections.values()]
+        connections.forEach((connection, index) => {
```
Comment (harbu) [inline test/integration/ping-pong.test.js:117]: Might be worth checking `expect(connections.length).toEqual(3)` just to be on the safe side.

File: test/integration/ping-pong.test.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            if (index === 0) {
+                expect(connection.isAlive)
+                    .toBeFalsy()
+            } else {
+                expect(connection.isAlive)
+                    .toBeTruthy()
+            }
+        })
+
+        client1.on('disconnected', () => {
```
Comment (harbu) [inline test/integration/ping-pong.test.js:128]: I don't understand why this and then `websocketServer._pingConnections()` at end of this test is needed.

Comment (mirpo) [reply test/integration/ping-pong.test.js:126]: - on first call of `_pingConnections` we set `respondedPong=false` for each websocket
- then we expect to get `pong` frame and set `respondedPong=true`
- after `pingInterval` seconds, we call again `_pingConnections`
- if `respondedPong=false`, it means that websocket didn't send back `pong` frame -> disconnect not active `websocket`
- if `respondedPong=true`  -> we got `pong` frame, everything is good, repeat from first step

**Pull request merged**